src/compiler.rs

```diff
@@ -39,6 +39,37 @@ macro_rules! expect {
     };
 }
 
+macro_rules! parse_branch {
+    ($compiler:expr, $block:expr, [ $( $call:expr ),* ]) => {
+        let block_length = $block.ops.len();
+        let token_length = $compiler.curr;
+        let num_errors = $compiler.errors.len();
+        let mut stored_errors = Vec::new();
+        let mut success = false;
+        // We risk getting a lot of errors if we are in an invalid state
+        // when we start the parse.
+        while !$compiler.panic {
+            $(
+                $call;
+                if !$compiler.panic {
+                    success = true;
+                    break;
+                }
+                $compiler.panic = false;
+                $compiler.curr = token_length;
+                let thrown_errors = $compiler.errors.len() - num_errors - 1;
```

Comment:
Isn't `thrown_errors` re-declared in the same scope if multiple `$call` are passed? Unless this macro expansion is automatically scoped (which is kinda cool).

Reply:
It is re-declared, but Rust allows that, shadowing the previous one. So the result is correct.

Reply:
Indeed